stan/math/prim/mat/fun/mdivide_left_tri.hpp

```diff
@@ -91,14 +91,10 @@ inline Eigen::Matrix<double, R1, C2> mdivide_left_tri(
 #ifdef STAN_OPENCL
   if (A.rows()
       >= opencl_context.tuning_opts().tri_inverse_size_worth_transfer) {
-    matrix_cl A_cl(A);
+    matrix_cl A_cl(A, TriView == Eigen::Lower ? TriangularViewCL::Lower
```

Comment:
This is much nicer to read than the separate calls, thanks. 
Do you think making a "conversion function" would be viable here, something like https://codereview.stackexchange.com/a/24199

On paper this seems nice for readability reasons. So the call would be matrix_cl A_cl(A, Eigen::Lower) and it would be converted to the TriangularViewCL enum.

Could we actually extract the view type inside the constructor from the Eigen matrix? For instance we could check if the matrix is Eigen::Lower or Eigen::Upper? Or could that cause any issues? We would have the option of specifying an override if needed. Do you think this would be of any value?

Reply:
conversion function seems like a good idea here. 

> Could we actually extract the view type inside the constructor from the Eigen matrix?

Eigen::Matrix does not contain information on triangular view. Alternatively we could change the function signature to accept Eigen::TriangularView. In this case the call will look like `mdivide_left_tri(A.triangularView<Eigen::Upper>, B)`